Sources/SwiftLanguageService/ParametersDocumentationExtractor.swift

```diff
@@ -0,0 +1,192 @@
+import Foundation
+import Markdown
+
+private struct ParametersDocumentationExtractor {
+  private var parameters = [String: String]()
```

Comment:
Instead of modifying a (global-ish, at least shared within this struct) member inside the struct, I wonder if it would be easier to read and debug if each of the `extract` returned the parameters that it extracted from the markdown it was given. That would follow the paradigm of immutable data structures more closely.